docs/scripting-ref.rst

```diff
@@ -346,30 +346,47 @@ a table to an Array use :ref:`treat-as-array`.
 
 JavaScript → Lua conversion rules:
 
-==============  =================
-JavaScript      Lua
-==============  =================
-string          string
-number          number
-boolean         boolean
-Object          table
-Array           table, marked as array (see :ref:`treat-as-array`)
-``undefined``   ``nil``
-``null``        ``""`` (an empty string)
-Date            string: date's ISO8601 representation, e.g. ``1958-05-21T10:12:00Z``
-function        an empty table ``{}`` (don't rely on it)
-==============  =================
-
-Function arguments and return values are passed by value. For example,
-if you modify an argument from inside a JavaScript function then the caller
-Lua code won't see the changes, and if you return a global JS object and modify
-it in Lua then object won't be changed in webpage context.
+================  =================
+JavaScript        Lua
+================  =================
+string            string
+number            number
+boolean           boolean
+Object            table
+Array             table, marked as array (see :ref:`treat-as-array`)
+``undefined``     ``nil``
+``null``          ``""`` (an empty string)
+Date              string: date's ISO8601 representation, e.g.
+                  ``1958-05-21T10:12:00.000Z``
+function          ``nil``
+circular object   ``nil``
+host object       ``nil``
+================  =================
 
 .. note::
 
     The rule of thumb: if an argument or a return value can be serialized
     via JSON, then it is fine.
 
+Note that currently you can't return DOM Elements, JQuery $ results and
+similar structures from JavaScript to Lua; to pass data you have to
+extract their attributes of interest as plain strings/numbers/objects/arrays:
```

Comment:
Yeah, Splash now doesn't return circular objects; instead of circular object detection there is now a depth limit. I went with depth limit because JS engine in Webkit doesn't have a Map object which can be used for detection of duplicate object references. Even https://github.com/douglascrockford/JSON-js/blob/master/json2.js uses O(N) lookups for cycle detection, which makes serialization O(N^2), I can't stand it :)

Another backwards-incompatible change is that previously Splash tried to serialize e.g. DOM elements with all their properties; it worked for a single element on a simple page, but didn't work if there are many of them. Now it fails fast and only supports explicit objects/arrays. 

It was not possible to return functions even before this PR - they were returned just as empty tables which is not useful. 

Another backwards-incompatible change is that splash:evaljs error info is slightly changed ('type' value is upper-cased).